x/sequencer/keeper/sequencer.go

```diff
@@ -19,14 +18,13 @@ func (k Keeper) SetSequencer(ctx sdk.Context, sequencer types.Sequencer) {
 
 	seqByRollappKey := types.SequencerByRollappByStatusKey(sequencer.RollappId, sequencer.SequencerAddress, sequencer.Status)
 	store.Set(seqByRollappKey, b)
+}
 
-	// To support InitGenesis scenario
-	if sequencer.Status == types.Unbonding {
-		k.setUnbondingSequencerQueue(ctx, sequencer)
-	}
+func (k Keeper) UpdateSequencer(ctx sdk.Context, sequencer types.Sequencer) {
+	k.UpdateSequencerWithStateChange(ctx, sequencer, sequencer.Status)
 }
 
-func (k Keeper) UpdateSequencer(ctx sdk.Context, sequencer types.Sequencer, oldStatus types.OperatingStatus) {
+func (k Keeper) UpdateSequencerWithStateChange(ctx sdk.Context, sequencer types.Sequencer, oldStatus types.OperatingStatus) {
```

Comment:
`UpdateSequencerWithStateChange` - does `state` here means `status` ? if so i think better change it.
Also if I think it's cleaner to have the function interface having oldStatus and newSTatus and not assume caller will change it on the object itself (cause otherwise it's not necessarily only updating the status but possibly other fields)